java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/Current.java

```diff
@@ -251,7 +251,7 @@ private OutgoingResponse createOutgoingResponseCore(Throwable exc) {
             replyStatus = ReplyStatus.UserException;
 
             if (requestId != 0) {
-                ostr.writeByte(replyStatus.value());
+                ostr.writeByte((byte) replyStatus.value());
```

Comment:
Shouldn't we use the generated code to marshal this?

Reply:
I'll rework this code when I implement DispatchException in Java.

Then, because enums are "checked" in Java, we can't use the generated code to marshal/unmarshal any ReplyStatus in Java. We can use the generated code to marshal known reply statuses.